File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    let answer = []
    let map = new Map();
    for (let i=0;i<s.length;i++) {
        if (map.has(s[i])) {
            answer[i] = i-map.get(s[i]) 
        } else {
            answer.push(-1)
        }
        map.set(s[i],i)
    } 
    return answer
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    last_seen = {}
    for i, char in enumerate(s):
        if char in last_seen:
            distance = i - last_seen[char]
            answer.append(distance)
        else:
            answer.append(-1)
            
        last_seen[char] = i

    return answer